packages/app/components/Today/ProfessorCheckinButton.tsx

```diff
@@ -0,0 +1,75 @@
+import { API } from "@koh/api-client";
+import { Modal, Radio } from "antd";
+import { useRouter } from "next/router";
+import React, { ReactElement, useState } from "react";
+import styled from "styled-components";
+import { useCourse } from "../../hooks/useCourse";
+import { useProfile } from "../../hooks/useProfile";
+import TACheckinButton, { CheckinButton } from "./TACheckinButton";
+
+const ProfessorModalRadio = styled(Radio)`
+  display: block;
+  height: 30px;
+  lineheight: 30px;
+`;
+
+export default function ProfessorCheckinButton(): ReactElement {
+  const profile = useProfile();
+  const [modalVisible, setModalVisible] = useState(false);
+  const [value, setValue] = useState(0);
+  const router = useRouter();
+  const { cid } = router.query;
+  const { course } = useCourse(Number(cid));
+  const queueCheckedIn = course?.queues.find((queue) =>
+    queue.staffList.find((staff) => staff.id === profile?.id)
+  );
+
+  return (
+    <>
+      {modalVisible && (
+        <Modal
+          title="Check-In To Office Hours"
+          visible={modalVisible}
+          onCancel={() => setModalVisible(false)}
+          okText="Check In"
+          onOk={async () => {
+            const redirectID = await API.taStatus.checkIn(
+              Number(cid),
+              course?.queues[value].room
+            );
+
+            router.push(
+              "/course/[cid]/queue/[qid]",
+              `/course/${Number(cid)}/queue/${redirectID.id}`
+            );
+          }}
+        >
+          <h3>Which queue would you like to check into?</h3>
+          <Radio.Group value={value} onChange={(e) => setValue(e.target.value)}>
+            {course?.queues.map((q, i) => (
+              <ProfessorModalRadio key={q.id} value={i}>
+                {q.room}
+              </ProfessorModalRadio>
+            ))}
+          </Radio.Group>
+        </Modal>
+      )}
+      {!queueCheckedIn && (
+        <CheckinButton
+          type="default"
+          size="large"
+          onClick={() => setModalVisible(true)}
+        >
+          Check In
+        </CheckinButton>
+      )}
+      {queueCheckedIn && (
+        <TACheckinButton
+          courseId={Number(cid)}
+          room={queueCheckedIn.room}
+          state="CheckedIn"
+        />
```

Comment:
Not a big deal but it could be nice to have a single `CheckinButton` that get's used in `today.tsx` which handles all the check-in/check-out logic for TA's and Professors

Reply:
I think I could manage that